torch_frame/data/dataset.py

```diff
@@ -707,20 +708,25 @@ def get_split(self, split: str) -> Dataset:
             raise ValueError(
                 f"'get_split' is not supported for '{self}' since 'split_col' "
                 f"is not specified.")
-        if split not in ["train", "val", "test"]:
-            raise ValueError(f"The split named '{split}' is not available. "
-                             f"Needs to be either 'train', 'val', or 'test'.")
         indices = self.df.index[self.df[self.split_col] ==
                                 SPLIT_TO_NUM[split]].tolist()
         return self[indices]
 
-    def split(self) -> tuple[Dataset, Dataset, Dataset]:
-        r"""Splits the dataset into training, validation and test splits."""
-        return (
-            self.get_split("train"),
-            self.get_split("val"),
-            self.get_split("test"),
-        )
+    def split(self) -> tuple[Dataset, Dataset, Dataset | None]:
+        r"""Splits the dataset into training, validation and optionally
+        test splits.
+        """
+        train_set = self.get_split(TrainingStage.TRAIN)
+        val_set = self.get_split(TrainingStage.VAL)
+        test_set = self.get_split(TrainingStage.TEST)
+        if test_set.num_rows == 0:
+            test_set = None
+        return train_set, val_set, test_set
+
+    def random_split(self, ratios: list[float] | None = None):
+        split = generate_random_split(self.num_rows, ratios)
+        self.split_col = 'split'
+        self.df[self.split_col] = split
```

Comment:
I think it'd be nice to supply an argument to control the random seed of the random seed.

Reply:
Also, add doc-string so that people know how to use it.